src/arg_suite/arg_suite.py

```diff
@@ -0,0 +1,298 @@
+# flake8: noqa
+# Do NOT modify. To be replaced by an eternal dependency
+"""Command-Line Arguments Parsing Suite
+
+This module is an command-line argument library that:
+
+    - generates argparse.ArgumentParser based a NamedTuple classed argument
+    - handles two-way conversions: a typed argument object (A NamedTuple) <-> a command-line argv
+    - enables IDE type hints and code auto-completion
+    - promotes type-safety of command-line arguments
+
+
+The following is a simple usage example that sums integers from the
+command-line and writes the result to a file::
+
+
+    @arg_suite
+    class MyArg(NamedTuple):
+        '''MyArg docstring goes to description
+        Each named (without "_" prefix) and fully typed attribute will be turned into an ArgumentParser.add_argument
+        '''
+
+        nn: List[int]  # Comments go to ArgumentParser argument help
+        _nn = {'choices': (200, 300)}  # (Advanced feature) Optional user supplement/override for ArgumentParser.add_argument("--nn", **_nn)
+
+        a_tuple: Tuple[str, int]  # Arguments without defaults are treated as "required = True"
+        h_param: Dict[str, int]  # Also supports List, Set
+
+        ###### arguments without defaults go first as required by NamedTuple ######
+
+        l_r: float = 0 # Arguments with defaults are treated as "required = False"
+        n: Optional[int] = None
+
+
+The module contains the following public classes:
+
+- arg_suite -- The main entry point for command-line argument suite. As the
+    example above shows, this decorator will attach an ArgSuite instance to
+    the argument NamedTuple "subclass".
+
+- ArgSuite -- The main class that generates the corresponding ArgumentParser
+    and handles the two-way conversions.
+
+All other classes and methods in this module are considered implementation details.
+"""
+
+__all__ = ('arg_suite', 'ArgSuite')
+import inspect
+import re
+import tokenize
+from argparse import *
+import logging
+from typing import *
+
+from typing_inspect import get_args, get_origin
```

Comment:
i tried running the script and got error due to `ModuleNotFoundError: No module named 'typing_inspect'`

do we need this as an dependency?

Reply:
`get_args, get_origin` part of later python. the autotest is passing, but not sure if that's set up to covers all the language that we want to support. what's detext's target python version?

Reply:
I was using python3.7.7. I need to install `typing-inspect` and the examples works fine. We should add to the setup.py as a dependency?

Reply:
depends. if we just want to support 3.7 and up, I might be able to update the code to not to rely the dependencies.

Reply:
Internally we're always using 3.7 and tests here are setup with 3.7.  I'm fine either way.

Reply:
removed dependency and added phyton 3.7 to setup.py, since it looks like we only test on 3.7.

Reply:
Thanks.